### PR TITLE
People: fix unknown prop JS warnings

### PR DESCRIPTION
**Pull request by roccotripaldi**
The intent of this PR is to fix JS warnings emitted in the People section.
While I was at it, I also fixed ES6 lint warnings in the affected files.

To reproduce:
- run master branch at calypso.localhost:3000
- go to /people/team/$site
- note the JS warnings emitted by `Card` and `PeopleProfile`
- go to /people/edit/$user/$site
- note the JS warnings emitted by `PeopleProfile`, `FormSelect`, and `RoleSelect`
- go to /people/new/$site
- note the JS warnings emitted by `FormSelect`

To test:
- checkout this branch
- ensure that the above JS errors are gone

Note:
This PR does not address the `valueLink` errors. That's a whole different ball of wax that can be addressed in an other PR.

cc: @ebinnion @tyxla @johnHackworth 
Test live: https://calypso.live/?branch=fix/unknown-prop-warnings-people

### CODE REVIEW

File: client/my-sites/people/invite-people/index.jsx [hunk 1]
```diff
@@ -361,7 +359,8 @@ const InvitePeople = React.createClass( {
 									disabled={ this.state.sendingInvites } />
 								<FormSettingExplanation>
 									{ this.translate(
-										'(Optional) You can enter a custom message of up to 500 characters that will be included in the invitation to the user(s).'
+										'(Optional) You can enter a custom message of up to 500 characters ' +
```
Comment (tyxla) [inline client/my-sites/people/invite-people/index.jsx]: I'd use a template literal here. That would also apply to the multiline text node within the other `FormSettingExplanation`  occurrence above.

**Comment by tyxla** [comment]
Works great, all of the mentioned notices are removed with the PR. 

With the exception of a very minor nitpicky line comment I've left, everything LGTM.

File: client/my-sites/people/invite-people/index.jsx (hunk 1 above)
Comment (gziolo) [inline client/my-sites/people/invite-people/index.jsx]: Interesting idea. Can you keep alignment as it is for multiline text without side effects?
At least on Mozilla site (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals), they start next line from the beginning of the line.

Comment (tyxla) [inline client/my-sites/people/invite-people/index.jsx]: I don't think there will be side effects in this case, but in general I think you're right - considering this it's probably not worth using template literals here, as side effect might come at some point.

Comment (gziolo) [inline client/my-sites/people/invite-people/index.jsx]: I'd still investigate further that idea ;)
@akirk would that be possible to use multiline translation with template string literal as proposed by @tyxla?

Comment (akirk) [inline client/my-sites/people/invite-people/index.jsx]: Yeah, please don't use template literals for this. The way it is split into multiple lines right now is the way to go. See https://github.com/Automattic/eslint-plugin-wpcalypso/blob/master/docs/rules/i18n-no-newlines.md

Thanks for the cc!

**Pull request merged**